tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
-
     # change logon drive to C:
     send_key_until_needlematch 'yast2_samba-server_netbios_logon-drive', 'down';
     wait_screen_change { send_key 'ret' };
-    wait_screen_change { send_key 'alt-l' };
-    for (1 .. 5) { send_key 'backspace'; }
-    type_string 'C:';
+    for (1 .. 2) { send_key 'backspace'; }
+    type_string($samba_directives->{logon_drive});
     wait_screen_change { send_key 'alt-o' };
     wait_screen_change { send_key 'alt-o' };
 
-    # swith to Trusted Domains
-    # add a trusted domain
-    send_key 'alt-t';
+    ## swith to Trusted Domains
+    # add a trusted domain, wait while all data is properly loaded
+    wait_still_screen(2);
+    wait_screen_change { send_key 'alt-t' };
     assert_screen 'yast2_samba-server_trusted-domains';
     wait_screen_change { send_key 'alt-a' };
-    type_string 'suse.de';
+    type_string('suse.de');
     wait_screen_change { send_key 'alt-p' };
-    type_string 'testing';
+    type_string('testing');
     send_key 'alt-o';
     assert_screen 'yast2_samba-server_trusted-domains_error';
     wait_screen_change { send_key 'alt-o' };
-
     # cancel trusted domain configuration
     wait_screen_change { send_key 'alt-c' };
 
-}
-
-sub samba_ldap_config {
-    # swith to LDAP Settings
-    send_key 'alt-l';
-    assert_screen 'yast2_samba-server_ldap-settings';
-    wait_screen_change { send_key 'alt-b' };
-    assert_screen 'yast2_samba-server_ldap_value_rewritten';
-    wait_screen_change { send_key 'alt-y' };
-    wait_screen_change { send_key 'alt-e' };
-    type_string 'ldap://localhost:389';
-
-    # set admin password and search base dn
-    wait_screen_change { send_key 'alt-a' };
-    type_string 'cn=admin,dc=qa,dc=suse,dc=de';
-    wait_screen_change { send_key 'alt-p' };
-    type_string 'testing';
-    wait_screen_change { send_key 'alt-g' };
-    type_string 'testing';
-    wait_screen_change { send_key 'alt-n' };
-    type_string 'dc=qa, dc=suse, dc=de';
-
-    # check advanced settings befor run test connection to ldap server
-    wait_screen_change { send_key 'alt-v' };
-    send_key 'ret';
-
-    # enter expert ldap settings
-    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings';
-
-    # change replication sleep and time-out
-    wait_screen_change { send_key 'alt-p' };
-    for (1 .. 10) { send_key 'down'; }
-    wait_screen_change { send_key 'alt-t' };
-    for (1 .. 2) { send_key 'up'; }
-
-    # change to not use SSL or TLS
-    wait_screen_change { send_key 'alt-u' };
-    for (1 .. 2) { send_key 'up'; }
-    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings_not-use-ssl';
-    wait_screen_change { send_key 'ret' };
-    wait_screen_change { send_key 'alt-o' };
-
-    # now run test connection
-    send_key 'alt-t';
-    assert_screen 'yast2_samba-server_ldap_test-connection';
-    wait_screen_change { send_key 'alt-o' };
+    if ((is_sle(">=15")) || (is_leap(">=15.0")) || is_tumbleweed) {
```

Comment:
would you mind turning this around? The approach I would like to follow in general is that code states the current as-is behaviour for openSUSE Tumbleweed which should be other reference and everything else is the exception. This also should simplify removing behaviour-branches for older versions, e.g. when the go EOL